authority-portal-frontend/src/app/pages/catalog-page/asset-detail-dialog/asset-not-available-dialog/asset-not-available-dialog.component.ts

```diff
@@ -0,0 +1,24 @@
+import {Component, OnDestroy} from '@angular/core';
+import {MatDialogRef} from '@angular/material/dialog';
+import {Subject} from 'rxjs';
+
+@Component({
+  selector: 'app-asset-not-available-dialog',
+  templateUrl: './asset-not-available-dialog.component.html',
+})
+export class AssetNotAvailableDialogComponent implements OnDestroy {
+  constructor(
+    private dialogRef: MatDialogRef<AssetNotAvailableDialogComponent>,
+  ) {}
+
+  closeDialog(): void {
+    this.dialogRef.close();
+  }
+
+  ngOnDestroy$ = new Subject();
+
+  ngOnDestroy(): void {
+    this.ngOnDestroy$.next(null);
+    this.ngOnDestroy$.complete();
+  }
```

Comment:
I'm pretty sure this is necessary because the class implements the `OnDestroy` interface. When opening the dialog, we subscribe to the destruction event to revert the URL back to root. -> `catalog-page.component.ts` in line 173